ckanext/canada/logic.py

```diff
@@ -529,3 +532,64 @@ def canada_datastore_run_triggers(up_func, context, data_dict):
         context['connection'] = backend._get_write_engine().connect()
     with datastore_create_temp_user_table(context, drop_on_commit=False):
         return up_func(context, data_dict)
+
+
+@chained_action
+@side_effect_free
+def portal_sync_package_show(up_func, context, data_dict):
+    """
+    Wraps package_show to customize the return of the Portal Sync harvest source package.
+    """
+    package = up_func(context, data_dict)
+
+    if package.get('type') == 'harvest':
+        if package.get('id') != PORTAL_SYNC_ID:
+            # we only allow the single harvest source
+            raise ObjectNotFound
+        else:
+            # custom return dict for our harvest source
+            _extras = {}
+            for _extra in package.get('extras', []):
+                _extras[_extra.get('key')] = _extra.get('value')
+            return {
+                'id': package.get('id'),
+                'name': package.get('name'),
+                'title': _(package.get('title')),
+                'source_type': package.get('source_type', _extras.get('source_type', None)),
+                'type': package.get('type'),
+                'url': package.get('url'),
+                'source': package.get('source', _extras.get('source', None)),
+                'target': package.get('target', _extras.get('target', None)),
+                'metadata_created': package.get('metadata_created'),
+                'metadata_modified': package.get('metadata_modified'),
+                'private': package.get('private'),
+                'state': package.get('state'),
+                'status': package.get('status'),
+                'resources': [],  # required for core
+            }
+
+    return package
```

Comment:
what's going on here? This won't prevent creation of harvest packages or prevent them showing up in search results.

Reply:
My bad, I think that was some crazy thing I was initially trying to do. But that chained action is not used so will remove.

The harvest sources NOT showing up in search results is handled by the ckanext-harvest plugin.